tap_salesforce/client.py

```diff
@@ -187,9 +190,29 @@ def get_records(
 
         query = f"{select_stm}{from_stm}{where_stm}{order_by_stm}{limit_stm}"
 
-        yield from self._paginate(
-            "GET", f"/services/data/{self._API_VERSION}/queryAll/", params={"q": query}
-        )
+        try:
+            yield from self._paginate(
+                "GET", f"/services/data/{self._API_VERSION}/queryAll/", params={"q": query}
+            )
+        except requests.exceptions.HTTPError as e:
+            nth = (end_date - start_date).total_seconds() / shrink_window_factor
+
+            # minimum allowed window size to get_records from before raising error...
+            if nth < timedelta(days=1).seconds:
```

Comment:
Minimum window, 1 day for now. I'm unsure if this can become a problem... Companies might have too much data in only one day? Maybe.

What is the optimal number here? 1 second, 1 minute, 1 hour .... 🤔 

@paddie @wizhi @hanssonandreas